worker/socialPoster.js

```diff
@@ -92,21 +92,32 @@ async function getHottestItem ({ models }) {
   return item[0]
 }
 
-async function itemToMessage ({ item }) {
+async function itemToMessage ({ item, postAuthorNostrProfile }) {
   return `${item.title}
 
-by ${item.userName} in ~${item.subName}
+by ${postAuthorNostrProfile ? `nostr:${postAuthorNostrProfile}` : `${item.userName}`} in ~${item.subName}
 ${numWithUnits(msatsToSats(item.msats), { abbreviate: false })} and ${numWithUnits(item.ncomments, { abbreviate: false, unitSingular: 'comment', unitPlural: 'comments' })} so far
 
 https://stacker.news/items/${item.id}`
 }
 
 export async function postToSocial ({ models }) {
   const item = await getHottestItem({ models })
-  if (item) {
-    const message = await itemToMessage({ item })
-    console.log('Message:', message)
-    await postToTwitter({ message })
-    await postToNostr({ message })
-  }
+  if (!item) return
+
+  const postAuthor = await models.user.findUnique({
+    where: { id: item.userId, hideNostr: false },
+    select: { nostrPubkey: true }
```

Comment:
You're going to want to also check `nostrAuthPubkey` ... sorry, we have two ways of adding a nostr pubkey.